addons/mail/static/src/core/common/thread_model.js

```diff
@@ -873,19 +887,26 @@ export class Thread extends Record {
         ]);
     }
 
-    markAsRead() {
+    /**
+     * @param {Object} [options]
+     * @param {boolean} [options.sync] Whether to sync the unread message
+     * state with the server values.
+     */
+    markAsRead({ sync } = {}) {
```

Comment:
This feels more natural for `markAsRead` to sync by default. Explicit `sync: false` justifies better the intent